### PR TITLE
[fix](Nereids) fix cast string to date

### PR DESCRIPTION
**Pull request by LiBinfeng-01**
### What problem does this PR solve?

Issue Number: close #xxx

Related PR: #35637 

Problem Summary:
When cast("201-01-01" as datetimev2(0)), The result is "2020-01-01" but it is wrong. It should be result in "0201-01-01".
201 would be regarded as 20xy-0z as related pr show, it was a bug.  But actually it should not have this trasformation and result in 

### Release note

None

### Check List (For Author)

- Test 
    - [x] Regression test
    - [ ] Unit Test
    - [ ] Manual test (add detailed scripts or steps below)
    - [ ] No need to test or manual test. Explain why:
        - [ ] This is a refactor/code format and no logic has been changed.
        - [ ] Previous test can cover this change.
        - [ ] No code files have been changed.
        - [ ] Other reason 

- Behavior changed:
    - [x] No.
    - [ ] Yes. 

- Does this need documentation?
    - [ ] No.
    - [ ] Yes. 

### Check List (For Reviewer who merge this PR)

- [ ] Confirm the release note
- [ ] Confirm test cases
- [ ] Confirm document
- [ ] Add branch pick label

### CODE REVIEW

**Comment by Thearas** [comment]
Thank you for your contribution to Apache Doris.
Don't know what should be done next? See [How to process your PR](https://cwiki.apache.org/confluence/display/DORIS/How+to+process+your+PR).

Please clearly describe your PR:
1. What problem was fixed (it's best to include specific error reporting information). How it was fixed.
2. Which behaviors were modified. What was the previous behavior, what is it now, why was it modified, and what possible impacts might there be.
3. What features were added. Why was this function added?
4. Which code was refactored and why was this part of the code refactored?
5. Which functions were optimized and what is the difference before and after the optimization?

**Comment by LiBinfeng-01** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run buildall

**Comment by LiBinfeng-01** [comment]
run p0

**Review by yagagagaga — APPROVED** [review]

**Review by englefly — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateLiteral.java [hunk 1]
```diff
@@ -189,19 +189,23 @@ static Result<String, AnalysisException> normalize(String s) {
...
                     sb.append(s, i, j);
                 } else if (len == 3) {
-                    if (partNumber == 0) {
-                        String yy = s.substring(i, i + 2);
-                        int year = Integer.parseInt(yy);
-                        if (year >= 0 && year <= 69) {
-                            sb.append("20");
-                        } else if (year >= 70 && year <= 99) {
-                            sb.append("19");
+                    if (s.charAt(j) == '.') {
```
Comment (englefly) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateLiteral.java:192]: why "201.01.01" is year 2020, but "201-01-01" is year 201

**Review by englefly — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateLiteral.java (hunk 1 above)
Comment (LiBinfeng-01) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/literal/DateLiteral.java:192]: this behavior is align to mysql

**Pull request merged**